### PR TITLE
Enable bodyclose linter

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?
Enable [bodyclose](https://github.com/timakin/bodyclose) linter: checks whether HTTP response body is closed successfully

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Review by ahmed-mez — APPROVED** [review]
Thank you!

**Review by brycekahle — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by nizox — APPROVED** [review]

File: cmd/trace-agent/test/runner.go [hunk 1]
```diff
@@ -179,6 +179,9 @@ func (s *Runner) DoReq(url, method string, payload []byte) (*http.Response, erro
 
 func (s *Runner) doRequest(req *http.Request) error {
```
Comment (katiehockman) [inline cmd/trace-agent/test/runner.go:180]: I wonder if this code should look like this instead:

```
func (s *Runner) doRequest(req *http.Request) error {
	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		return err
    }
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusOK {
		slurp, err := ioutil.ReadAll(resp.Body)
		if err != nil {
			return fmt.Errorf("%s (error reading response body: %v)", resp.Status, err)
		}
		return fmt.Errorf("%s: %s", resp.Status, slurp)
	}
	return nil
}
```

The [docs](https://pkg.go.dev/net/http#Client.Do) say "On error, any Response can be ignored" and yet we are trying to read the response body and status code without first checking that `Do` returned a nil error. That seems like a separate bug here. I'm happy to change that in a separate PR if you don't want to mess with that here though.

**Review by IvanTopolcic — APPROVED** [review]

**Review by mbotarro — APPROVED** [review]

**Review by djmitche — COMMENTED** [review]
for pkg/flare/flare.go

**Review by djmitche — APPROVED** [review]

**Review by katiehockman — APPROVED** [review]

File: cmd/trace-agent/test/runner.go (hunk 1 above)
Comment (katiehockman) [reply cmd/trace-agent/test/runner.go:180]: Created https://github.com/DataDog/datadog-agent/pull/12850

**Pull request merged**